neptune/internal/storage/storage_utils.py

```diff
@@ -95,8 +101,22 @@ def __repr__(self):
 
 def split_upload_files(upload_entries, max_package_size=1 * 1024 * 1024, max_files=500):
     current_package = UploadPackage()
+    walked_entries = set()
     for entry in upload_entries:
-        size = 0 if os.path.isdir(entry.source_path) else os.path.getsize(entry.source_path)
+        if os.path.isdir(entry.source_path):
+            for root, _, files in os.walk(entry.source_path):
+                path_relative_to_entry_source = os.path.relpath(root, entry.source_path)
+                if path_relative_to_entry_source == ".":
```

Comment:
We can replace this if/else with
```py
target_root = os.path.join(entry.target_path, path_relative_to_entry_source)
normalized_root = os.path.normpath(target_root)
```

This should be less sensitive to different platforms.